mlperf_logging/package_checker/seed_checker.py

```diff
@@ -0,0 +1,148 @@
+import warnings
+import os
+
+from ..compliance_checker import mlp_parser
+
+# What are source files?
+SOURCE_FILE_EXT = {
+    '.py', '.cc', '.cpp', '.cxx', '.c', '.h', '.hh', '.hpp', '.hxx', '.sh',
+    '.sub', '.cu', '.cuh'
+}
+
+
+def is_source_file(path):
+    """ Check if a file is considered as a "source file" by extensions.
+
+    The extensions that are considered as "source file" are listed in
+    SOURCE_FILE_EXT.
+
+    Args:
+        path: The absolute path, relative path or name to/of the file.
+    """
+    return os.path.splitext(path)[1].lower() in SOURCE_FILE_EXT
+
+
+def find_source_files_under(path):
+    """ Find all source files in all sub-directories under a directory.
+
+    Args:
+        path: The absolute or relative path to the directory under query.
+    """
+    source_files = []
+    for root, subdirs, files in os.walk(path):
+        for file_name in files:
+            if is_source_file(file_name):
+                source_files.append(os.path.join(root, file_name))
+    return source_files
+
+
+class SeedChecker:
+    """ Check if the seeds fit MLPerf submission requirements.
+    Current requirements are:
+
+    1. All seeds must be logged through mllog (if choose to log seeds). Any seed
```

Comment:
Some typos to fix: Disgarded --> Discarded, least --> least.

Reply:
`Disgarded` fixed.
I'm wondering what's wrong with `least` ?

Reply:
Sorry, lesat --> least on (3)